web/src/lib/nostr/utils.ts

```diff
@@ -183,3 +171,15 @@ export async function setPublicKey(user: User | null) {
 
     return true;
 }
+
+export function getKeyFromKeyOrNpub(key: string) {
+    if (key.toLowerCase().startsWith("npub")) {
+        let decoded = nip19.decode(key);
+        if (decoded.type !== "npub" || typeof decoded.data !== 'string') {
```

Comment:
Good catch! I'll just remove the 2nd check for "npub" actually.

The thing is, I copied this code from another place, where I didn't do the first check, so it made sense to check  that what nostr-tools found is indeen an npub in that case. But now the 1st check is necessary (because I want to know whether to even try to decode an npub, or assume it is hex...) - so the 2nd check is superfluous. Proper validation (like the length) - *should* be nostr-tools responsibility. And if the length is incorrect but it starts with npub, then ... I should still treat it as npub rather than skip decoding and assume it is hex...